src/reducers/profile-view.js

```diff
@@ -169,6 +169,35 @@ function viewOptionsPerThread(state: ThreadViewOptions[] = [], action: Action) {
         ...state.slice(threadIndex + 1),
       ];
     }
+    case 'CHANGE_INVERT_CALLSTACK': {
+      const { callTree, callNodeTable, selectedThreadIndex } = action;
+      return state.map((viewOptions, threadIndex) => {
+        let selectedCallNodePath;
+        let expandedCallNodePaths;
+        if (selectedThreadIndex === threadIndex) {
+          // Only attempt this on the current thread, as we need the transformed thread
+          // There is no guarantee that this has been calculated on all the other threads,
+          // and we shouldn't attempt to expect it, as that could be quite a perf cost.
+          selectedCallNodePath = Transforms.invertCallNodePath(
+            viewOptions.selectedCallNodePath,
+            callTree,
+            callNodeTable
+          );
+
+          expandedCallNodePaths = [];
+          for (let i = 1; i < selectedCallNodePath.length; i++) {
+            expandedCallNodePaths.push(selectedCallNodePath.slice(0, i));
+          }
+        } else {
+          selectedCallNodePath = [];
+          expandedCallNodePaths = [];
```

Comment:
I believe we shouldn't change these values in this case. We should just keep the previous values.

Try the following STR:
1. select a node
2. click on a different thread (from its title, not its timeline, so that we don't select a row by mistake)
3. invert the call tree
4. uninvert the call tree
5. come back on the initial thread

=> everything is collapsed with this patch, it wasn't before.

Reply:
Yeah, I was concerned about having incorrect data. This could be solved by doing per-thread inversion, which is out of scope of this PR. The current behavior is to have incorrect information, so I agree with you on this, although I think in the future we should figure out a better solution.